src/BuildScriptGenerator/Python/PythonPlatform.cs

```diff
@@ -243,7 +243,12 @@ public PlatformDetectorResult Detect(RepositoryContext context)
                     = compressedVirtualEnvFileName;
             }
 
-            this.TryLogDependencies(pythonVersion, context.SourceRepo);
+            var customRequirementsTxtPath = this.pythonScriptGeneratorOptions.CustomRequirementsTxtPath;
+            var requirementsTxtPath = customRequirementsTxtPath == null ? PythonConstants.RequirementsFileName : customRequirementsTxtPath;
+            if (context.SourceRepo.FileExists(requirementsTxtPath))
+            {
+                this.TryLogDependencies(requirementsTxtPath, pythonVersion, context.SourceRepo);
+            }
```

Comment:
Here I'm asking for a behavioral change, so feel free to ignore if you believe that would be too risky. Would it still make sense to throw an exception if the env variable was set but that the file doesn't exist? We could argue that if a customer set the environment variable, then he/she WANTS this file to be taken into account, and that it would be better to fail early than just silently ignoring this.

Reply:
Discussed offline -- moved to throw an exception when `CUSTOM_REQUIREMENTSTXT_PATH` is provided but isn't a valid path to fail-fast for the user and given them an opportunity to update the environment variable to the correct path, or scrap it altogether.